### PR TITLE
[QAA-173]Speculos x LLM (local test)

### PR DESCRIPTION
**Pull request by abdurrahman-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached. (no need)
- [X] **Covered by automatic tests.** 
- [X] **Impact of the changes:** 
  - No regression on current LLM mocked tests

### 📝 Description

_First Speculos Add account test. Works only locally, another ticket will be done for the CI_

Precondition : 

- Have a speculos image tagged "speculos"

- Export following envVars :

```
export SPECULOS_IMAGE_TAG=speculos
export SEED="YOUR SEED"
export COINAPPS="PATH to the FOLDER"
```

- Build App

```
pnpm i --filter="live-mobile..." --filter="ledger-live" --filter="@ledgerhq/dummy-*-app...
pnpm build:llm:deps
pnpm build:dummy-apps
pnpm mobile e2e:build -c ios.sim.release
```
- Launch test :

```
pnpm mobile e2e:test -c ios.sim.release --testMatch $(pwd)/apps/ledger-live-mobile/e2e/specs/speculos/**
```

### ❓ Context

- **JIRA or GitHub link**: 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by ypolishchuk-ledger — DISMISSED** [review]

File: apps/ledger-live-mobile/e2e/bridge/proxy.ts [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+const job = ({ device, port, silent, verbose, speculosApiPort }: ProxyOptions) =>
+  new Observable(observer => {
+    const req: SpeculosHttpTransportOpts = {
+      apiPort: speculosApiPort,
+    };
+
+    transport = {
+      id: `speculos-http-${speculosApiPort}`,
+      open: id =>
+        id.includes(port) ? retry(() => retry(() => SpeculosHttpTransport.open(req))) : null,
```
Comment (valpinkman) [inline apps/ledger-live-mobile/e2e/bridge/proxy.ts:94]: [ASK] do you need 2 `retry()` here ?

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount.spec.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+describe("Add accounts", () => {
+  beforeAll(async () => {
+    app = await Application.init("onboardingcompleted");
+    await app.portfolio.waitForPortfolioPageToLoad();
+  });
+
+  currencies.forEach(({ currency, nanoApp, tmsLink }) => {
+    let deviceName: string;
+
+    $TmsLink(tmsLink);
```
Comment (valpinkman) [inline apps/ledger-live-mobile/e2e/specs/speculos/addAccount.spec.ts:19]: seems to be specific to IntelliJ Idea, are we sure it works correctly with other IDE/Code editors ?

**Review by valpinkman — COMMENTED** [review]
Amazing that we can use speculos in our 2e2 tests 🚀

**Review by valpinkman — CHANGES_REQUESTED** [review]
see previous comment 🙏

File: apps/ledger-live-mobile/e2e/specs/speculos/addAccount.spec.ts (hunk 2 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/addAccount.spec.ts:19]: Not sure but I am using VS Code and it is working for me. I referred the `jest-allure2-reporter` plugin documentation [here](https://wix-incubator.github.io/jest-allure2-reporter/docs/features/links/#tms-links)

File: apps/ledger-live-mobile/e2e/bridge/proxy.ts (hunk 1 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/bridge/proxy.ts]: You are right, I removed one.

**Review by ypolishchuk-ledger — APPROVED** [review]

**Review by valpinkman — APPROVED** [review]

**Pull request merged**